test/s3path.jl

```diff
@@ -421,6 +421,31 @@ function s3path_tests(config)
         end
     end
 
+    @testset "isdir" begin
+        # Top level
+        path = S3Path("s3://$(bucket_name)"; config=config)
+        @test isdir(path) == true
+
+        # No Such Bucket
+        path = S3Path("s3://some_non_existent_bucket_7051649213"; config=config)
```

Comment:
We should make this an actual AWSException w/ the `NoSuchBucket` error so we're not dependent on someone not making this an actual bucket.